lib/model/unittest/CCountingModelTest.cc

```diff
@@ -137,14 +138,10 @@ BOOST_FIXTURE_TEST_CASE(testCheckScheduledEvents, CTestFixture) {
     factory.features(features);
 
     {
-        CModelFactory::SGathererInitializationData gathererNoGapInitData(startTime);
-        CModelFactory::TDataGathererPtr gatherer(factory.makeDataGatherer(gathererNoGapInitData));
-        CModelFactory::SModelInitializationData modelNoGapInitData(gatherer);
-        this->addArrival(*gatherer, 200, "p");
-
-        CAnomalyDetectorModel::TModelPtr modelHolderNoGap(factory.makeModel(modelNoGapInitData));
-        CCountingModel* modelNoGap =
-            dynamic_cast<CCountingModel*>(modelHolderNoGap.get());
+        this->makeModel(params, features, startTime);
+        CCountingModel* modelNoGap = dynamic_cast<CCountingModel*>(m_Model.get());
+        BOOST_TEST_REQUIRE(modelNoGap);
+        BOOST_REQUIRE_EQUAL(std::size_t(0), this->addPerson("p", m_Gatherer));
```

Comment:
```suggestion
        BOOST_REQUIRE_EQUAL(0, this->addPerson("p", m_Gatherer));
```

There's no need to cast expected types with Boost.Test.  The places where we do this in existing code are a hangover from CppUnit.